spanner/src/enable_fine_grained_access.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/main/spanner/README.md
+ */
+
+namespace Google\Cloud\Samples\Spanner;
+
+// [START spanner_enable_fine_grained_access]
+use Google\Cloud\Spanner\Admin\Database\V1\DatabaseAdminClient;
+use \Google\Cloud\Iam\V1\Binding;
+use \Google\Type\Expr;
+
+/**
+ * Enable Fine Grained Access.
+ * Example:
+ * ```
+ * enable_fine_grained_access($projectId, $instanceId, $databaseId, $databaseRole, $iamMember);
+ * ```
+ *
+ * @param string $projectId The Google cloud project ID
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ * @param string $databaseRole The database role bound to
+ * the IAM member.
+ * @param string $iamMember The IAM member.
+ * Eg: `user:{emailid}`, `serviceAccount:{emailid}`, `group:{emailid}`, `domain:{domain}`
+ */
+function enable_fine_grained_access(
+    string $projectId,
+    string $instanceId,
+    string $databaseId,
+    string $databaseRole,
+    string $iamMember
+    ): void {
+    $adminClient = new DatabaseAdminClient();
+    $resource = sprintf('projects/%s/instances/%s/databases/%s', $projectId, $instanceId, $databaseId);
+    $policy = $adminClient->getIamPolicy($resource);
+
+    if ($policy->getVersion() != 3) {
+        $policy->setVersion(3);
+    }
+
+    $binding = new Binding([
+        'role' => 'roles/spanner.fineGrainedAccessUser',
+        'members' => [$iamMember],
+        'condition' => new Expr([
+            'title' => 'DatabaseRoleBindingTitle',
+            'expression' => sprintf("resource.name.endsWith('/databaseRoles/%s')", $databaseRole)
+            ])
+    ]);
+    $policy->setBindings([$binding]);
+    $adminClient->setIamPolicy($resource, $policy);
+
+    printf('Enabled fine-grained access in IAM.' . PHP_EOL);
```

Comment:
Usually we don't add full stop into print statements:

```suggestion
    printf('Enabled fine-grained access in IAM' . PHP_EOL);
```

Reply:
Done